pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -356,11 +356,16 @@ func doInstall(ctx *zedmanagerContext,
 		}
 	}
 
-	// Any StorageStatus to add?
+	// Any StorageStatus to add or need to resolve tags?
 	for _, sc := range config.StorageConfigList {
 		ss := lookupStorageStatus(status, sc)
 		if ss != nil {
```

Comment:
I wouldn't do this in this "add" code unless there is a very good reason.
I don't know if there can be cases where the storagestatus has already been added but the resolution isn't done.

Thus putting the checks in the subsequent loop before we check
		if !ss.HasVolumemgrRef {

and have the added check be
			if ss.IsContainer && !ss.ResolveDone {
			ss.HasResolverRef = true
			MaybeAddResolveConfig(ctx, ss)
			// XXX state name could be "waiting for tag resolutions(s)" but
			// painful to introduce a new state end to end
			minState = types.INITIAL
			ss.State = types.INITIAL
                        changed = true
                        continue
}

Note that you want to loop over all the storage status hence the continue is important.

Note that the way we normally do this is we pick up the error in this code (from VolumeStatus, AppNetworkStatus, DomainStatus) and I think that pattern is needed here as well so you can add the clearing of errors (following the IsErrorSource patern we have for the others).
That way the downloader can retry a failed (timed out) resolution and it will recover. Hence on line 119 in handleResolveStatusModify you would just call doUpdate for the app instance, and the error as well as sha extraction would be in this doUpdate code.

(And please create a story to have downloader retry a failed resolution. Can implement that separately.)

Reply:
>I don't know if there can be cases where the storagestatus has already been added but the resolution isn't done.

While creating an instance, we will have storage status without resolving tags because handle create parses over the storage config list and adds storage status before calling doUpdate.

Reply:
Sure, but there can be updates for other volumes so doUpdate() might be called and it will continue since it doesn't check if you're waiting for the resolver.
doUpdate() needs to check that everything is in place before proceeding to the next step.
And if you do that you might as well do the work in doUpdate to make this be uniform with the handling of the other *Status. Makes for less unique behaviors and bugs for the resolver interaction.

Reply:
I updated the PR.